aten/src/ATen/native/quantized/cpu/qnnpack/src/qnnpack/pack_block_sparse.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+#include <cstdint>
+#include <vector>
+
+#include <qnnpack/math.h>
+
+typedef struct {
+  std::vector<uint32_t> col_indices;
+  std::vector<uint32_t> row_values;
+  std::vector<uint8_t> values;
+  uint32_t col_block_size;
+  uint32_t row_block_siz{1};
+  void print() {
+    std::cout << "row ptr\n";
+    for (const auto& t : row_values) {
+      std::cout << t << ", ";
+    }
+    std::cout << std::endl;
+    std::cout << "col indices\n";
+    for (const auto& t : col_indices) {
+      std::cout << t << ", ";
+    }
+    std::cout << std::endl;
+    std::cout << "Actual values\n";
+    for (const auto& t : values) {
+      std::cout << (uint32_t)t << ", ";
+    }
+    std::cout << std::endl;
+  }
+} BCSRMatrix;
+
+BCSRMatrix generateBlockCSRMatrix(
+    const uint8_t* a,
+    const size_t N,
+    const size_t K,
+    const uint32_t col_block_size,
+    const uint8_t* zero_points) {
+  BCSRMatrix bcsr_mat;
+  bcsr_mat.row_values.resize(N);
+  // K must be > 0
```

Comment:
An assert would be helpful to check for preconditions.  I think generous use of asserts is good practice - has no runtime cost but can come in handy when dealing with code that behaves in unexpected ways.